subprojects/docs/src/docs/userguide/jvm/toolchains.adoc

```diff
@@ -15,31 +15,36 @@
+Compile tasks may use `javac` as their compiler, test and exec tasks may use the `java` command while `javadoc` will be used to generate documentation.
+
+By default, Gradle uses the same Java toolchain for running Gradle itself and building JVM projects.
+However, this may not be always desirable.
 Building projects with different Java versions on different developer machines and CI servers may lead to unexpected issues.
 Additionally, you may want to build a project using a Java version that is not supported for running Gradle.
 
-A Java Toolchain (from now on referred to simply as toolchain) is a set of tools, usually taken from a local JRE/JDK installation that are used to configure different aspects of a build.
-Compile tasks may use `javac` as their compiler, test and exec tasks may use the `java` command while `javadoc` will be used to generate documentation.
+In order to improve reproducibility of the builds and make build requirements clearer, Gradle allows to configure toolchains on both project and task level.
 
 [[sec:consuming]]
-== Consuming Toolchains
+== Toolchains for projects
 
-A build can globally define what toolchain it targets by stating the Java Language version it needs and optionally the vendor:
+A build can globally define what toolchain it targets by stating the Java language version it needs:
 
 ====
-include::sample[dir="samples/java/jvm-multi-project-with-toolchains/kotlin/",files="buildSrc/src/main/kotlin/myproject.java-conventions.gradle.kts[tags=toolchain]"]
-include::sample[dir="samples/java/jvm-multi-project-with-toolchains/groovy/",files="buildSrc/src/main/groovy/myproject.java-conventions.gradle[tags=toolchain]"]
+include::sample[dir="snippets/java/toolchain-basic/kotlin",files="build.gradle.kts[tags=toolchain]"]
+include::sample[dir="snippets/java/toolchain-basic/groovy",files="build.gradle[tags=toolchain]"]
 ====
 
 Executing the build (e.g. using `gradle check`) will now handle several things for you and others running your build:
 
-1. Setup all compile, test and javadoc tasks to use the defined toolchain which may be different than the one Gradle itself uses
-2. Gradle detects <<#sec:auto_detection,locally installed JVMs>>
-3. Gradle chooses a JRE/JDK matching the requirements of the build (in this case a JVM supporting Java 11)
-4. If no matching JVM is found, it will automatically download a matching JDK based on the <<#sub:download_repositories,toolchain download repositories>> configured.
+1. Gradle configures all compile, test and javadoc tasks to use the defined toolchain.
+2. Gradle detects <<#sec:auto_detection,locally installed toolchains>>.
+3. Gradle chooses a toolchain matching the requirements (any Java 17 toolchain for the example above).
+4. If no matching toolchain is found, Gradle will automatically download a matching one based on the configured <<#sub:download_repositories,toolchain download repositories>>.
```

Comment:
I would replace "will" with "can" since in Gradle 8, this requires to opt-in by adding and configuring a settings plugin